From estarkov@ru.mvista.com Fri Jan 11 14:54:21 2008
Date: Thu, 20 Dec 2007 17:15:38 +0300
From: Egor Starkov <estarkov@ru.mvista.com>
To: mingo@elte.hu
Cc: rostedt@goodmis.org, linux-rt-users@vger.kernel.org,
     linux-kernel@vger.kernel.org, linuxppc-dev@ozlabs.org
Subject: PPC64 doesn't compile with CONFIG_SMP=n

Hello Ingo

I've found out that real-time tree doesn't compile for PPC64 with 
CONFIG_SMP=n.

Think this is due to patch-2.6.21.4-rt10 patch. It has
definitions of following symbols missing: __get_cpu_lock,
__get_cpu_var_locked.

I've attached the patch to fix the problem.

Egor Starkov

    [ Part 2: "Attached Text" ]

Signed-off-by: Egor Starkov <estarkov@ru.mvista.com>

---
 include/asm-powerpc/percpu.h |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-2.6.24-rt1/include/asm-powerpc/percpu.h
===================================================================
--- linux-2.6.24-rt1.orig/include/asm-powerpc/percpu.h	2008-01-25 15:07:37.000000000 -0500
+++ linux-2.6.24-rt1/include/asm-powerpc/percpu.h	2008-01-25 15:08:33.000000000 -0500
@@ -68,6 +68,8 @@ extern void setup_per_cpu_areas(void);
 
 #define __get_cpu_var(var)			per_cpu__##var
 #define __raw_get_cpu_var(var)			per_cpu__##var
+#define __get_cpu_lock(var, cpu)		per_cpu_lock__##var##_locked
+#define __get_cpu_var_locked(var, cpu)		per_cpu__##var##_locked
 
 #endif	/* SMP */
 
